src/main.rs

```diff
@@ -1,26 +1,63 @@
-fn main() {
-    loop {
-        println!("Olá deseja adicionar um novo ToDo?");
-        println!("[sim/nao]");
-        let ask = input();
-        if ask == "sim" {
-            println!("Digite um novo ToDo abaixo:");
-            let todo = input();
-            println!("     ");
-            println!("{} // Adicionado com sucesso! //", todo);
-        } else if ask == "nao" {
-            println!("Encerrando programa em");
-            println!("3");
-            println!("2");
-            println!("1");
-            break
+#[derive(Debug, Clone)]
+struct Todo {
+    message: String
+}
+
+impl Todo {
+    fn new(message: String) -> Self{
+        Todo {
+            message
+        }
+    }
+}
+use std::io::{Stdin, Stdout, Write};
+
+struct Terminal {
+    stdin: Stdin,
+    stdout: Stdout,
+}
+
+impl Terminal {
+    fn new() -> Self {
+        Terminal {
+            stdin: std::io::stdin(),
+            stdout: std::io::stdout()
         }
     }
+
+    fn input(&mut self) -> String {
+        let mut buf: String = String::new();
+        self.stdin.read_line(&mut buf).unwrap();
+        buf.trim().to_string()
+    }
+
+    fn ask_for_new_todo(&mut self) -> Todo {
+        println!("Olá deseja adicionar um novo ToDo? (Digite 'sim' para adicionar) ");
+        
+        let res = self.input();
+        if res.to_lowercase() == "sim" {
+            println!("Digite o ToDo que deseja criar: ");
+            let todo_res = self.input();
+            print!("Todo adicionado 👍 : ");
+            let todo: Todo = Todo::new(todo_res);
```

Comment:
Em geral, você só tipa explicitamente uma variável quando o compilador não é capaz de inferir o tipo que você quer.

```suggestion
            let todo = Todo::new(todo_res);
```